library-core/src/main/kotlin/elements/events.kt

```diff
@@ -1,16 +1,17 @@
 package elements
 
 import com.google.gson.JsonElement
-import com.pusher.platform.BaseClient
+import com.pusher.platform.network.parseAs
+import com.pusher.platform.network.parseOr
 
 sealed class SubscriptionMessage {
     companion object Factory {
         fun fromRaw(messageString: String): SubscriptionMessage {
-            val message = BaseClient.GSON.fromJson(messageString, Array<JsonElement>::class.java)
+            val message = messageString.parseAs<Array<JsonElement>>()
```

Comment:
What if this is not valid json?

Reply:
Very true, I'll make sure it returns a `Result` instead :)